packages/starrating/src/react/index.js

```diff
@@ -0,0 +1,155 @@
+import * as glamor from 'glamor'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import { withTheme } from '@pluralsight/ps-design-system-theme/react'
+
+import Star from './star'
+
+import css, { BASE_CLASSNAME } from '../css'
+
+const styles = {
+  starRating: () => glamor.css(css[BASE_CLASSNAME]),
+  screenReaderInput: () =>
+    glamor.css(css[`${BASE_CLASSNAME}__screen-reader-input`]),
+  screenReaderText: () =>
+    glamor.css(css[`${BASE_CLASSNAME}__screen-reader-text`])
+}
+
+const ScreenReaderInput = props => (
+  <input {...styles.screenReaderInput(props)} tabIndex={-1} {...props} />
+)
+
+const ScreenReaderText = props => (
+  <span {...styles.screenReaderText(props)} {...props} />
+)
+
+class StarRating extends React.PureComponent {
+  constructor(props) {
+    super(props)
+
+    this.handleStarClicked = this.handleStarClicked.bind(this)
+    this.handleStarEnter = this.handleStarEnter.bind(this)
+    this.handleStarLeave = this.handleStarLeave.bind(this)
+
+    this.state = { hoverIndex: null, interactive: !!this.props.onChange }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const interactive = !!nextProps.onChange
+    if (interactive === this.props.interactive) return
+
+    this.setState({ interactive })
+  }
+
+  handleInputChanged(event) {
+    if (!this.state.interactive) return
+    this.props.onChange(event.target.value, event)
+  }
+
+  handleStarClicked(index, event) {
+    if (!this.state.interactive) return
+    this.props.onChange(index + 1, event)
+  }
+
+  handleStarEnter(index, event) {
+    if (!this.state.interactive) return
+    this.setState(() => ({ hoverIndex: index }))
+  }
+
+  handleStarLeave(index, event) {
+    if (!this.state.interactive) return
+    this.setState(() => ({ hoverIndex: null }))
+  }
+
+  get stars() {
```

Comment:
I don't use getters often.  Teach me, please, why do you reach for a getter in this case?

Reply:
the render method was getting messy and i wanted to encapsulate some rendering logic without creating a new component. i generally reach for a getter when i need this but i also get weird looks :)